src/components/activity/screens/ActivityList/actions/index.js

```diff
@@ -34,29 +34,52 @@ export function getActivityList() {
+      let promises = [];
+      request = `SELECT id, title, description, activity_date, content_source, type, interlocutor_id
       FROM activity
-      WHERE title LIKE ? OR description LIKE ?`,
-      ["%" + request + "%","%" + request + "%"]
-    )
-      .then((res) => {
-        const activityListFromRequest = res.rows._array.map(activity => ({
-          activity: {
-            id: activity.id,
-            title: activity.title,
-            description: activity.description,
-            activity_date: activity.activity_date,
-            content_source: activity.content_source,
-            type: activity.type,
-            interlocutor_id: activity.interlocutor_id,
-          }
-        }));
-        dispatch({ type: GET_ACTIVITY_LIST_FROM_REQUEST, activityListFromRequest });
+      WHERE title LIKE ?
+      OR description LIKE ?`;
+      requestList.forEach((term) => {
+        if (term.length > 0) {
+          promises = promises.concat(executeSql(request, [`%${  term  }%`, `%${  term  }%`]));
+        }
       });
+      return Promise.all(promises)
+        .then((res) => {
+          let activityListFrequency = {}; // Key : activity_id ; Value : [activity object, frequency]
+          res.forEach((r) => {
+            r.rows._array.forEach((activity) => {
+              if (activity.id in activityListFrequency) {
+                activityListFrequency[activity.id][1] += 1;
+              } else {
+                activityListFrequency[activity.id] = [{
+                  activity: {
+                    id: activity.id,
+                    title: activity.title,
+                    description: activity.description,
+                    activity_date: activity.activity_date,
+                    content_source: activity.content_source,
+                    type: activity.type,
+                    interlocutor_id: activity.interlocutor_id,
+                  },
+                },1];
+              }
+            });
+          });
+          // Create an array based on the dictionnary activityListFrequency
+          const activityListFrequencyArray = Object.keys(activityListFrequency).map(key => [key, activityListFrequency[key][0], activityListFrequency[key][1]]);
+          let activityListFromRequest = [];
+
+          // Fill the activityListFromRequest by a list of tuple (activity object, frequency)
+          activityListFrequencyArray.forEach((activity) => {
+            activityListFromRequest = activityListFromRequest.concat([[activity[1],activity[2]]]);
+          });
+          dispatch({ type: GET_ACTIVITY_LIST_FROM_REQUEST, activityListFromRequest });
+        });
+    };
   }
-  else {
-    activityListFromRequest = []
-    return (dispatch) => dispatch({ type: GET_ACTIVITY_LIST_FROM_REQUEST, activityListFromRequest });
-  }
-}
\ No newline at end of file
+  const activityListFromRequest = [];
+  return dispatch => dispatch({ type: GET_ACTIVITY_LIST_FROM_REQUEST, activityListFromRequest });
```

Comment:
Same remak as in src/components/interlocutor/screens/InterlocutorList/actions/index.js : double dispatch